File: 0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int minp=Integer.MAX_VALUE;
        int maxp=0;

        for(int price:prices){
            if(price<minp){
                minp=price;
            }else if(price-minp>maxp){
                maxp=price-minp;
            }
        }
        return maxp;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minp = float('inf') 
        maxp = 0

        for price in prices:
            if price < minp:
                minp = price
            elif price - minp > maxp:
                maxp = price - minp

        return maxp

        
